### PR TITLE
Replaced SODA API with MongoDB API

### PR DESCRIPTION
**Pull request by msupic**
To enable 'Oracle Database API for MongoDB' ([requirements](https://docs.oracle.com/en/cloud/paas/autonomous-database/adbsa/mongo-using-oracle-database-api-mongodb.html)) I had to:

- modify Network Access Type to 'Secure access from allowed IPs and VCNs only' for autonomous database (used 'oke_vcn')
- add 'Web Access' to the mushop_carts_user (added CONNECT and RESOURCE roles to the user and enabled REST calls using ORDS_ADMIN.ENABLE_SCHEMA)
- expose autonomous database host via 'oadb-connection' kubernetes secret and ORACLECLOUD_ATP_HOST env variable to the Carts Pod so mongodb url can be created

I also upgraded version of micronaut data module to use micronaut serialization/deserialization which removed native image issues related to codecs.

Note: still need to test deployment of entire mushop app to AWS which will be separate task

### CODE REVIEW

File: src/carts/pom.xml [hunk 1]
```diff
@@ -242,6 +197,11 @@
             <plugin>
                 <groupId>io.micronaut.build</groupId>
                 <artifactId>micronaut-maven-plugin</artifactId>
+                <configuration>
+                    <nativeImageBuildArgs>
+                        <nativeImageBuildArg>-H:AdditionalSecurityProviders=com.sun.security.sasl.Provider</nativeImageBuildArg>
```
Comment (msupic) [inline src/carts/pom.xml:202]: Still need to investigate whether this can be added to mongodb micronaut module

File: src/carts/pom.xml [hunk 2]
```diff
@@ -27,13 +27,12 @@
         <release.version>11</release.version>
         <exec.mainClass>mushop.carts.Application</exec.mainClass>
         <micronaut.runtime>netty</micronaut.runtime>
-        <micronaut.version>3.5.3</micronaut.version>
-        <micronaut.data.version>3.5.0</micronaut.data.version>
-        <micronaut.oraclecloud.version>2.1.5</micronaut.oraclecloud.version>
+        <micronaut.version>3.6.3</micronaut.version>
```
Comment (alvarosanchez) [inline src/carts/pom.xml:30]: Now that 3.7.0 is out, perhaps it's an opportunity to not be outdated again.

File: src/carts/pom.xml [hunk 3]
```diff
@@ -27,13 +27,12 @@
...
+        <micronaut.data.version>3.7.3</micronaut.data.version>
         <micronaut.micrometer.version>4.3.0</micronaut.micrometer.version>
```
Comment (alvarosanchez) [inline src/carts/pom.xml:32]: Can this be removed too and use whatever version comes from the BOM?

File: src/carts/pom.xml [hunk 4]
```diff
@@ -27,13 +27,12 @@
...
         <jdbc.version>21.1.0.0</jdbc.version>
-        <soda.version>1.1.7.1</soda.version>
         <docker.image.suffix>openjdk</docker.image.suffix>
+        <test.containers.version>1.17.2</test.containers.version>
```
Comment (alvarosanchez) [inline src/carts/pom.xml:35]: This is defined in the BOM as:

```xml
<testcontainers.version>1.17.3</testcontainers.version>
```

File: src/carts/pom.xml [hunk 5]
```diff
@@ -27,13 +27,12 @@
         <release.version>11</release.version>
         <exec.mainClass>mushop.carts.Application</exec.mainClass>
         <micronaut.runtime>netty</micronaut.runtime>
-        <micronaut.version>3.5.3</micronaut.version>
-        <micronaut.data.version>3.5.0</micronaut.data.version>
-        <micronaut.oraclecloud.version>2.1.5</micronaut.oraclecloud.version>
+        <micronaut.version>3.6.3</micronaut.version>
+        <micronaut.data.version>3.7.3</micronaut.data.version>
         <micronaut.micrometer.version>4.3.0</micronaut.micrometer.version>
         <jdbc.version>21.1.0.0</jdbc.version>
```
Comment (alvarosanchez) [inline src/carts/pom.xml:33]: Is this still needed?

File: src/carts/pom.xml [hunk 6]
```diff
@@ -118,53 +105,27 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongodb-driver-sync</artifactId>
         </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>mongodb</artifactId>
-            <version>1.16.3</version>
+            <version>${test.containers.version}</version>
```
Comment (alvarosanchez) [inline src/carts/pom.xml:119]: Remove the version

File: src/carts/src/main/java/mushop/carts/repositories/awsdocdb/AwsDocDbDefaultMongoConfiguration.java [hunk 7]
```diff
@@ -43,6 +48,7 @@
 /**
  * Overridden default MongoDB configuration class that replaces default codecs by the AWS DocDB specific ones.
  */
+@Requires(env="ec2")
```
Comment (alvarosanchez) [inline src/carts/src/main/java/mushop/carts/repositories/awsdocdb/AwsDocDbDefaultMongoConfiguration.java:51]: Better use the constant `io.micronaut.context.env.Environment.AMAZON_EC2`

**Review by alvarosanchez — CHANGES_REQUESTED** [review]
Great job here! 🚀

File: src/carts/pom.xml (hunk 3 above)
Comment (msupic) [reply src/carts/pom.xml]: Removed

File: src/carts/pom.xml (hunk 4 above)
Comment (msupic) [reply src/carts/pom.xml]: Removed

File: src/carts/pom.xml (hunk 5 above)
Comment (msupic) [reply src/carts/pom.xml]: Removed

File: src/carts/pom.xml (hunk 6 above)
Comment (msupic) [reply src/carts/pom.xml]: Removed

File: src/carts/src/main/java/mushop/carts/repositories/awsdocdb/AwsDocDbDefaultMongoConfiguration.java (hunk 7 above)
Comment (msupic) [reply src/carts/src/main/java/mushop/carts/repositories/awsdocdb/AwsDocDbDefaultMongoConfiguration.java]: Changed

**Review by alvarosanchez — APPROVED** [review]

**Comment by graemerocher** [comment]
Agreed, excellent work, well done @msupic !

**Pull request merged**